reactor-core/src/main/java/reactor/core/publisher/OnNextFailureStrategy.java

```diff
@@ -0,0 +1,193 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package reactor.core.publisher;
+
+import java.util.function.BiConsumer;
+import java.util.function.BiPredicate;
+import javax.annotation.Nullable;
+
+import org.reactivestreams.Subscription;
+import reactor.core.Exceptions;
+import reactor.util.context.Context;
+
+/**
+ * A strategy to deal with errors that happens during an operator's onNext call, by either
+ * propagating the error or allowing the sequence to continue. This opt-in strategy is
+ * applied by compatible operators through the {@link Operators#onNextFailure(Object, Throwable, Context, Subscription)}
+ * and {@link Operators#onNextPollFailure(Object, Throwable, Context)} methods.
+ * See {@link #STOP}, {@link #RESUME_DROP} and {@link #resume(BiConsumer)} for the
+ * possible strategies.
+ *
+ * @author Simon Baslé
+ */
+public interface OnNextFailureStrategy {
+
+	/**
+	 * The key that can be used to store an {@link OnNextFailureStrategy} in a {@link Context}.
+	 */
+	String KEY_ON_NEXT_ERROR_STRATEGY = "reactor.onNextError.localStrategy";
+
+	/**
+	 * Apply the {@link OnNextFailureStrategy} by returning a {@link Throwable} when the
+	 * error strategy doesn't allow sequences to continue, or null otherwise.
+	 *
+	 * @param value The onNext value that caused an error.
+	 * @param error The error.
+	 * @param context The most significant {@link Context} in case the strategy needs it.
+	 * @param forCancel The {@link Subscription} that should be cancelled if the
+	 * strategy is terminal. Null to ignore (typically from a poll()).
+	 * @param <T> The type of the value causing the error.
+	 * @return a {@link Throwable} to propagate through onError if the strategy is
+	 * terminal and cancelled the subscription, null if not.
+	 */
+	@Nullable
+	<T> Throwable apply(T value, Throwable error, Context context,
+			@Nullable Subscription forCancel);
+
+	/**
+	 * A terminal strategy where the error is passed through the {@link Operators#onOperatorError(Subscription, Throwable, Object, Context)}
+	 * hook (which therefore cancels the upstream subscription and propagates an error,
+	 * potentially different from the original one).
+	 */
+	OnNextFailureStrategy STOP = new OnNextFailureStrategy() {
+
+		@Override
+		public <T> Throwable apply(T value, Throwable error, Context context, Subscription forCancel) {
+			return Operators.onOperatorError(forCancel, error, value, context);
+		}
+
+	};
+
+	/**
+	 * A non-terminal strategy where the error is passed to the {@link Operators#onErrorDropped(Throwable, Context)}
+	 * hook and the incriminating source value is passed to the {@link Operators#onNextDropped(Object, Context)}
+	 * hook, allowing the sequence to continue with further values.
+	 */
+	OnNextFailureStrategy RESUME_DROP = new OnNextFailureStrategy() {
```

Comment:
I'd just use `RESUME` personally, because if the event gets dropped or not really depends on what the user does with it in the callbacks. If they send it to a DLQ for example, then it hasn't been dropped.  These strategy names should reflect the behaviour from the Flux perspective IMO.

Reply:
Ah, i see what you've done now, this is a built-in version of RESUME that just drop.  Needs a bit of cleanup because the strategy isn't really a strategy, and exposes constants/methods to implement different strategies.